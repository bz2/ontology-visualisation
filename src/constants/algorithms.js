export const ALGO_TYPE_FULL = 'full'
export const ALGO_TYPE_NEIGHBOURHOOD = 'neighbourhood'
export const ALGO_TYPE_SHORTEST_PATH = 'shortest-path'
export const ALGO_TYPE_BOUNDING_BOX = 'bounding-box'
export const ALGO_TYPE_NODES_FILTER = 'nodes-filter'
export const ALGO_TYPE_EDGES_FILTER = 'edges-filter'
