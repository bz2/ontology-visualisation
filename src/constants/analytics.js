export const USERBACK_ACCESS_TOKEN = '29230|42198|CVJlhtfkaYN3J3xP2x200ghJ1'
export const USERBACK_COOKIE = 'ubwc'
export const USERBACK_WIDGET = 'https://static.userback.io/widget/v1.js'
export const GA_ID = 'UA-195025859-1'
