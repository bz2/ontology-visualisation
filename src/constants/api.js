export const API_ENDPOINT = `${process.env.endpointUrl}/api` // 'http://localhost:8080/api'

export const GET_ONTOLOGY = `${API_ENDPOINT}/ontology`
export const GET_GRAPH = `${API_ENDPOINT}/graph`
export const GET_GRAPH_QUERY = `${GET_GRAPH}/query`

export const GET_EDGES = `${GET_GRAPH}/edges`
export const GET_EDGE = `${GET_GRAPH}/edges/{id}`

export const GET_NODES = `${GET_GRAPH}/nodes`
export const GET_NODE = `${GET_GRAPH}/nodes/{id}`

// export const GET_PROPERTIES = `${GET_GRAPH}/properties`
// export const GET_PROPERTIES_OBJECTS = `${GET_PROPERTIES}/objects`
// export const GET_PROPERTIES_OBJECT = `${GET_PROPERTIES}/objects/{id}`
// export const GET_PROPERTIES_ANNOTATIONS = `${GET_PROPERTIES}/annotations`
// export const GET_PROPERTIES_ANNOTATION = `${GET_PROPERTIES}/annotations/{id}`
export const GET_PROPERTIES = `${GET_ONTOLOGY}/properties`
export const GET_PROPERTIES_OBJECTS = `${GET_PROPERTIES}/objects`
export const GET_PROPERTIES_OBJECT = `${GET_PROPERTIES}/objects/{id}`
export const GET_PROPERTIES_ANNOTATIONS = `${GET_PROPERTIES}/annotations`
export const GET_PROPERTIES_ANNOTATION = `${GET_PROPERTIES}/annotations/{id}`

export const AUTH_SIGN_IN = `${API_ENDPOINT}/login`
