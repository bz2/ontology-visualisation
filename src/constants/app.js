export const APP_NAME = 'OntoPop'
