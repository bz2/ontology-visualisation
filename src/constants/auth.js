export const AUTH_COOKIE = 'ontology_viz_auth'
