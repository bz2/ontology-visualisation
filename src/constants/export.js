export const EXPORT_JPG = 'jpeg'
export const EXPORT_PNG = 'png'
export const EXPORT_PDF = 'pdf'
export const EXPORT_CSV = 'csv'
export const EXPORT_OWL = 'owl'

export const EXPORT_GRAPH_OPTIONS = [
  EXPORT_JPG,
  EXPORT_PNG,
  EXPORT_PDF
]

export const EXPORT_DATA_OPTIONS = [
  EXPORT_CSV,
  EXPORT_OWL
]
