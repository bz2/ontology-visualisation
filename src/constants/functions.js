export const getElementProperties = (element) => Object.keys(element).filter((elementKey) => typeof elementKey !== 'object')
export const getEdgeAndNodes = (edge) => edge.split('___')
export const getNodesInEdge = (edge) => getEdgeAndNodes(edge).slice(-2)
export const getEdgeUniqueId = (edge) => getEdgeAndNodes(edge)[0]
export const getPathEdges = (path) => path.split('|||')
export const toDashedCase = (text) => text.replace(/[A-Z]/g, '-$&').toLowerCase()
export const turnToRoute = (text) => `/app/${text}`
