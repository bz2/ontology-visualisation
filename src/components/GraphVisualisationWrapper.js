import { useEffect } from 'react'
import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import GraphVisualisation from './GraphVisualisation'
// import jsonClasses from '../assets/json/test-ontology-classes.json'
// import jsonObjectProperties from '../assets/json/test-ontology-object-properties.json'
import actions from '../store/actions'
import setNodesIdsToDisplay from '../utils/setNodesIdsToDisplay'
import { ALGO_TYPE_FULL } from '../constants/algorithms'
import getAllTriplesPerNode from '../utils/getAllTriplesPerNode'
import getGraphData from '../utils/getGraphData'

const GraphVisualisationWrapper = ({
  currentGraph,
  graphData,
  setStoreState,
}) => {
  const { t } = useTranslation()

  useEffect(async () => {
    const { classes, objectProperties } = await getGraphData({
      setStoreState,
      t
    })

    // Set data from local file for debugging
    // const classesFromApi = jsonClasses.OwlClasses
    // const objectPropertiesFromApi = jsonObjectProperties.OwlObjectProperties

    setStoreState('classesFromApi', classes)
    setStoreState('objectPropertiesFromApi', objectProperties)

    const classesIds = Object.keys(classes)
    const predicatesIds = Object.keys(objectProperties)

    // in the background, parse classes to get triples per node
    await getAllTriplesPerNode({
      classesIds,
      predicatesIds,
      setStoreState,
      classesFromApi: classes
    })

    setNodesIdsToDisplay({
      type: ALGO_TYPE_FULL,
      classesFromApi: classes,
      objectPropertiesFromApi: objectProperties,
      setStoreState
    })
  }, [])

  useEffect(() => {
    // Update nodes to display based on selected graph
    const {
      type,
      options
    } = graphData[currentGraph]

    setNodesIdsToDisplay({
      type,
      setStoreState,
      options
    })
  }, [
    currentGraph
  ])

  return (
    <GraphVisualisation />
  )
}

GraphVisualisationWrapper.propTypes = {
  currentGraph: PropTypes.string.isRequired,
  graphData: PropTypes.shape().isRequired,
  setStoreState: PropTypes.func.isRequired,
}

const mapToProps = ({
  currentGraph,
  graphData,
  isSettingsOpen,
}) => ({
  currentGraph,
  graphData,
  isSettingsOpen,
})

export default connect(
  mapToProps,
  actions
)(GraphVisualisationWrapper)
