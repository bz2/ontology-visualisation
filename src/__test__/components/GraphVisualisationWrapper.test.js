import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import GraphVisualisationWrapper from '../../components/GraphVisualisationWrapper'
import { ALGO_TYPE_FULL } from '../../constants/algorithms'

const setup = ({
  showContextMenu,
  isBoundingBoxSelectable
}) => {
  const props = {
    currentGraph: 'graph-0',
    graphData: {
      'graph-0': {
        label: 'Main',
        noDelete: true,
        type: ALGO_TYPE_FULL
      }
    },
    setStoreState: jest.fn(),
    addToObject: jest.fn(),
    showContextMenu,
    contextMenuData: {
      top: 0,
      left: 0,
      nodeId: ''
    },
    isBoundingBoxSelectable,
    boundingBoxGeometry: {
      fixedPointX: 100,
      fixedPointY: 100,
      boundingBoxPosX: 100,
      boundingBoxPosY: 100,
      boundingBoxWidth: 200,
      boundingBoxHeight: 200
    }
  }

  const component = shallow(<GraphVisualisationWrapper {...props} />)

  return {
    component,
    props
  }
}

describe('GraphVisualisationWrapper', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot when no context menu and no bounding box', () => {
    const {
      component
    } = setup({
      showContextMenu: false,
      isBoundingBoxSelectable: false
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when context menu and bounding box', () => {
    const {
      component
    } = setup({
      showContextMenu: true,
      isBoundingBoxSelectable: true
    })

    expect(toJson(component)).toMatchSnapshot()
  })
})
