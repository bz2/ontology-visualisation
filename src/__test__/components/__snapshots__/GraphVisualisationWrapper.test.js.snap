// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphVisualisationWrapper should match snapshot  1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
      "updateGraphData": [Function],
    }
  }
  currentGraph="graph-0"
  graphData={
    Object {
      "graph-0": Object {
        "label": "Main",
        "noDelete": true,
        "type": "full",
      },
    }
  }
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
>
  <Component />
</Connect>
`;
